src/integrationtests/java/com/aws/greengrass/integrationtests/ipc/IPCTestUtils.java

```diff
@@ -136,9 +139,18 @@ public static EventStreamRPCConnection connectToGGCOverEventStreamIPC(SocketOpti
 
         try (EventLoopGroup elGroup = new EventLoopGroup(1); ClientBootstrap clientBootstrap = new ClientBootstrap(elGroup, null)) {
 
-            String ipcServerSocketPath = kernel.getContext().get(IPCEventStreamService.class).getIpcServerSocketAbsolutePath();
+            //String ipcServerSocketPath = kernel.getContext().get(IPCEventStreamService.class).getIpcServerSocketPath();
+            final String[] ipcServerSocketPath = new String[1];
+            kernel.getConfig().getRoot().lookup(SETENV_CONFIG_NAMESPACE, NUCLEUS_DOMAIN_SOCKET_FILEPATH_FOR_COMPONENT)
+                    .subscribe(new Subscriber() {
+                                   @Override
+                                   public void published(WhatHappened what, Topic t) {
+                                       ipcServerSocketPath[0] = (String)t.getOnce();
+                                   }
+                               });
```

Comment:
This feels backwards. The IPC should be writing into SETENV, not reading from it.

Reply:
This is a method to test connecting to GGC through IPC. So it should be reading, right?

Reply:
OK sorry. Yep that's fine.